_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+
+Things that you can automatically test that help security:
+* Performance
+* Security headers
+* TLS and its configurations
+* Unit tests (also integration test, end-to-end tests, smoke tests and all the similar)
+* Known vulnerability analysis for your dependencies
+* Fuzz testing
+* Static code analysis for your source code
+* Best practice analysis in some cases
+
+You should not forget the manual testing. For example:
+* Review what others have done
+* Exploratory testing
+* Penetration testing
+* Trying out disaster recovery practices
+
+#### Deployment
+
+The moment you publish your source code as a software it becomes legacy and you need to support it. From SSDLC perspective this means things that you need actively do and things that you need to have processes for. 
+
+* Change management
+* Reading news (and vulnerability informations)
+* Incident and response
+* Monitoring and alerting
+
+#### Retirement
+
+This is often forgotten in all projects. At some day the time of our beloved system will come to an end. At this point the system has had a great deal or organical growth and understanding retirements implications can be hard to see. System needs to be taken down in a managed manner so that business can move to new world as it will but at the same time all the resources of the system needs to be disposed in a governed manner. Failing to due so will lead into information leak. 
+
+Things that you should consider:
+* Migration plan
+* Archiving data (there might be laws)
+* Destroying virtual or physical resources
+* Cleaning up the data stores
+* Cleaning all the DNS names and such to avoid subdomain takeovers (especially with cloud)
+
+
+#### Example implementation
+
+People at Unity Tech have been so awesomely kind that they have published their SSDLC as an [open source](https://github.com/UnityTech/unity-ssdlc/blob/master/Overview.md). I would strongly recommend looking at it as an example of an implementation.
+
+## My experience on the matter
+
+Taking security oriented tasks on the backlog has been natural since security has been brought up in front. Communication between different parties has been easy since we have had structure and documentation for the security conversations to take place. For example not a single threat modeling session has been in vain. It might feel before like we would already know everything but there is always something we had missed. 
```

Comment:
"It might feel before like we would already know everything but there is always something we had missed. " -> "It might feel before that we already know everything but there is always something we have missed.